### PR TITLE
Fix database commands and allow using OTD's in PMs.

### PR DESCRIPTION
**Pull request by HarmanPC**

### CODE REVIEW

**Comment by HarmanPC** [comment]
Sorry, I had closed previous pull request because I had deleted forked repo.

File: plugins/writing.js [hunk 1]
```diff
@@ -2124,32 +2127,36 @@ let commands = {
...
-			if (isNaN(Number(targets[1]))) return this.say("That was not an index number. Please use the number that's stated in the entry for the thing you're trying to edit.");
+			if (isNaN(Number(num))) return this.say("That was not an index number. Please use the number that's stated in the entry for the thing you're trying to edit.");
 			let pattern = /((http|https|ftp):\/\/)[^\s]/;
 			if (!pattern.test(targets[2])) {
 				return this.say("Please enter a valid URL.");
 			}
 			for (let i = 0; i < database.myths.db.length; i++) {
-				if (targets[1] === database.myths.db[i].id) {
-					database.myths.db[i].img = targets[2];
+				if (num >= database.myths.db.length) {
```
Comment (sirDonovan) [inline plugins/writing.js]: This check should go before the <code>for</code> loop.

File: plugins/writing.js [hunk 2]
```diff
@@ -2124,32 +2127,36 @@ let commands = {
...
+					return this.say('Cannot find entry number ' + num);
+				} else {
+					database.myths.db[num].img = targets[2];
```
Comment (sirDonovan) [inline plugins/writing.js]: You actually no longer need the <code>for</code> loop if you're doing it this way.

File: plugins/writing.js [hunk 3]
```diff
@@ -2124,32 +2127,36 @@ let commands = {
...
-			return this.say("Entry not found. Are you sure you're using the right myth index number?");
 		} else if (targets[0] === "remove" || targets[0] === "delete") {
 			if (room instanceof Users.User || !user.hasRank(room, '%')) return false;
-			if (isNaN(Number(targets[1]))) return this.say("That was not an index number. Please use the number that's stated in the entry for the thing you're trying to edit.");
+			let num = Tools.toId(targets[1]);
+			if (isNaN(Number(num))) return this.say("That was not an index number. Please use the number that's stated in the entry for the thing you're trying to edit.");
 			for (let i = 0; i < database.myths.db.length; i++) {
-				if (database.myths.db[i].id === targets[1]) {
-					database.myths.pendingDelete = Number(targets[1]);
+				if (num >= database.myths.db.length) {
```
Comment (sirDonovan) [inline plugins/writing.js]: Same comments as in <code>addimage</code>.

**Comment by HarmanPC** [comment]
@sirDonovan Sorry I don't understand this error.

File: plugins/writing.js [hunk 4]
```diff
@@ -684,6 +684,7 @@ let commands = {
 	'word': 'wotd',
 	wotd: function (target, room, user) {
 		let text = room instanceof Users.User || user.hasRank(room, '+') ? '' : '/pm ' + user.name + ', ';
+		if (room.id !== 'writing') return this.say('Please only use this command in Writing.');
```
Comment (sirDonovan) [inline plugins/writing.js]: You also need to check that the command is not being used in PMs.

File: plugins/writing.js [hunk 5]
```diff
@@ -794,25 +795,7 @@ let commands = {
...
-							this.say("Redeeming your Poetic License... Uses remaining: " + database.scribeShop[i].motw + "!");
-						}
-					}
-				}
-			}
-		} else if (!(room instanceof Users.User) && user.hasRank(room, '+')) {
-			hasPerms = true;
-		}
-		if (!hasPerms) return this.say(text + 'You must be at least Voice or higher to set the Myth of the Week.');
+		if (!user.hasRank(room, '+')) return this.say(text + 'You must be at least Voice or higher to set the Myth of the Week.');
```
Comment (sirDonovan) [inline plugins/writing.js]: This removes the ability to fix typos in PMs. You should put this in an <code>else</code> block under <code>if (targets[0] === "typo")</code>.

File: plugins/writing.js [hunk 6]
```diff
@@ -870,25 +853,7 @@ let commands = {
...
-							this.say("Redeeming your Poetic License... Uses remaining: " + database.scribeShop[i].wotd + "!");
-						}
-					}
-				}
-			}
-		} else if (!(room instanceof Users.User) && user.hasRank(room, '+')) {
-			hasPerms = true;
-		}
-		if (!hasPerms) return this.say(text + 'You must be at least Voice or higher to set the History of the Day.');
+		if (!user.hasRank(room, '+')) return this.say(text + 'You must be at least Voice or higher to set the History of the Day.');
```
Comment (sirDonovan) [inline plugins/writing.js]: Same issue with typos as above.

**Comment by sirDonovan** [comment]
@HarmanPC the error is a Typescript error. You can't use <code>user.hasRank</code> with a <code>user</code>, so you need to add <code>!(room instanceof Users.User)</code>.

File: plugins/writing.js [hunk 7]
```diff
@@ -794,25 +795,6 @@ let commands = {
 		if (database.motw) {
 			if (!typo && Date.now() - database.motw.time < 432000000) return this.say(text + "Sorry, but at least 5 days must have passed since the MOTW was last set in order to set it again!");
 		}
-		let hasPerms = false;
```
Comment (sirDonovan) [inline plugins/writing.js]: You should only be removing Scribe Shop stuff here, not the rank checking.

File: plugins/writing.js [hunk 8]
```diff
@@ -870,30 +852,11 @@ let commands = {
 		if (database.hotd) {
 			if (!typo && Date.now() - database.hotd.time < 61200000) return this.say(text + "Sorry, but at least 17 hours must have passed since the WOTD was last set in order to set it again!");
 		}
-		let hasPerms = false;
```
Comment (sirDonovan) [inline plugins/writing.js]: Same issue with rank checking.

**Comment by sirDonovan** [comment]
After you fix those, you also need to rebase with the master branch.

**Comment by HarmanPC** [comment]
ok if i did something wrong sorry because not much knowledge of git ;-;

**Comment by HarmanPC** [comment]
sorry again >_> closed by mistake while commenting

**Pull request closed without merging**